app/presenters/order_delivery_status_actions_presenter.rb

```diff
@@ -0,0 +1,21 @@
+class OrderDeliveryStatusActionsPresenter
+
+  def initialize(user, order, view_context)
+    @user = user
+    @order = order
+    @vc = view_context
+  end
+
+  def render
+    if Order::DeliveryStatusPolicy.new(user, order).mark_delivered?
+      vc.button_tag "Mark all delivered", class: 'pull-right btn btn--small btn--primary mobile-block', style: 'margin-left: 15px;', id: 'mark-all-delivered', type: 'button'
+    elsif Order::DeliveryStatusPolicy.new(user, order).mark_undelivered?
+      vc.button_tag "Undo mark delivery", class: 'pull-right btn btn--small btn--primary mobile-block' , id: 'undo-delivery', type: 'button'
+    end
```

Comment:
You are using both double and single quoting for strings in your arguments which is odd. I try to only use single quoting in ruby code unless I am doing interpolation, that way if I see a double quote I know that interpolation is happening in the string.

Reply:
Total mistake. I'm trying to use single quotes for non-interpolating strings but missed these.